src/native/managed/cdacreader/tests/CodeVersionsTests.cs

```diff
@@ -63,202 +64,80 @@ public static MockMethodDesc CreateVersionable (TargetPointer selfAddress, Targe
             };
         }
     }
+
     internal class MockCodeBlockStart
     {
         public TargetCodePointer StartAddress { get; set;}
         public uint Length { get; set; }
-
-        public bool Contains(TargetPointer ip) =>  ip >= StartAddress && ip < StartAddress + Length;
-        public bool Contains(TargetCodePointer ip) => Contains(ip.AsTargetPointer);
         public MockMethodDesc MethodDesc {get; set;}
-        public TargetPointer MethodDescAddress => MethodDesc.Address;
     }
 
-    internal class MockExecutionManager : IExecutionManager
-    {
-        private IReadOnlyCollection<MockCodeBlockStart> _codeBlocks;
+    private Mock<IRuntimeTypeSystem> mockRuntimeTypeSystem = new(MockBehavior.Strict);
```

Comment:
I tend to find it easier to reason about tests when everything is contained in the test method (explicitly sets up what it needs / no state outside the method, don't have to know that each test method actually gets a new test class instance).

Maybe try something like having the test cases explicitly create the mock(s) they need and turn the `Add*` methods into extension methods for `Mock<...>`?

Reply:
Happy to make this change. I settled on having a `MockExtensions` class which adds the helpers as extensions on the correct generic implementation of `Mock<...>`.